pkg/server/middleware/middlewares.go

```diff
@@ -60,20 +65,29 @@ func (b *Builder) BuildChain(ctx context.Context, middlewares []string) *alice.C
 
 		chain = chain.Append(func(next http.Handler) (http.Handler, error) {
 			constructorContext := internal.AddProviderInContext(ctx, middlewareName)
-			if _, ok := b.configs[middlewareName]; !ok {
+			if midInf, ok := b.configs[middlewareName]; !ok || midInf.Middleware == nil {
```

Comment:
```suggestion
			if b.configs[middlewareName] == nil {
```

Reply:
That suggestion changes the logic of that test. even if b.configs[middlewareName] exists, we also want to make sure that b.configs[middlewareName].Middleware exists too.